piksi_tools/settings.py

```diff
@@ -239,16 +245,20 @@ def write_from_file(self, output, verbose=False):
         return
 
     def _print_callback(self, msg, **metadata):
-        print(msg.text)
+        print(msg.text.decode('ascii'))
 
     def _settings_callback(self, sbp_msg, **metadata):
         section, setting, value, format_type = sbp_msg.payload.split(
-            '\0')[:4]
-        self.read_response_wait_dict[(section, setting)] = value
+            b'\0')[:4]
+        self.read_response_wait_dict[(
+            section.decode(KEY_ENCODING), setting.decode(KEY_ENCODING))] = value.decode(VALUE_ENCODING)
 
     def _settings_list_callback(self, sbp_msg, **metadata):
-        section, setting, value, format_type = sbp_msg.payload[2:].split(
-            '\0')[:4]
+        section_b, setting_b, value_b, format_type_b = sbp_msg.payload[2:].split(
+            b'\0')[:4]
```

Comment:
keep the statement on line line if possible

Reply:
the other occurrence becomes a slightly over-long line but there are also others anyway in the same file so probably it's neater your way. Done in the review fix commit.